### PR TITLE
dashboard: tls support

### PR DESCRIPTION
**Pull request by niedhui**
Signed-off-by: niedhui <niedhui@gmail.com>

### What problem does this PR solve? 

Add TLS support for the dashboard.  I'll update go.sum after https://github.com/pingcap-incubator/tidb-dashboard/pull/195 is accepted 

### Check List 

Tests 
 
 - Manual test (add detailed scripts or steps below)

Install client certificates for browser, then access https://127.0.0.1:2379 using the browser

### CODE REVIEW

**Comment by niedhui** [comment]
update `go.mod` since  https://github.com/pingcap-incubator/tidb-dashboard/pull/195 is merged 
@shafreeck PTAL, thanks

File: go.mod [hunk 1]
```diff
@@ -62,3 +62,5 @@ require (
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0
 	gopkg.in/yaml.v2 v2.2.8 // indirect
 )
+
+replace github.com/pingcap-incubator/tidb-dashboard v0.0.0-20200306154053-b5a3f9acae4f => github.com/niedhui/tidb-dashboard v0.0.0-20200310160535-08d1707aee36
```
Comment (nolouch) [inline go.mod]: remove replace.

Comment (niedhui) [reply go.mod]: Sorry. forgot this line :)

**Pull request merged**